Src/AutoNSubstitute/NSubstituteVirtualMethodsCommand.cs

```diff
@@ -204,9 +205,20 @@ private object Resolve(Type type)
                 // otherwise, NSubstitute would not be able to set up the methods
                 // that return a circular reference.
                 // See discussion at https://github.com/AutoFixture/AutoFixture/pull/397
-                return Task.Factory
-                    .StartNew(() => Context.Resolve(type))
-                    .Result;
+                var task = Task.Factory.StartNew(() => Context.Resolve(type));
+
+                // It could happen that task above is inlined on the current thread.
+                // As result, the last NSubstitute call router could become empty.
+                // Therefore, we pass cancellation token to the Wait() method to prevent task from being inlined.
```

Comment:
Wouldn't it be simpler to explicitly start a new thread, using `ThreadPool.QueueUserWorkItem` or `Thread.Start`?